### DISCUSSION TITLE
Blend modes support in Sprite3D

### DISCUSSION THREAD

**Discussion by jupiterbjy** [godotengine/godot-proposals#8418] [3D]
I can't find any details about blending modes in documents, nor I can find one in `Sprite3D`'s options, so assuming this feature not exists.

## Suggestion:

Add an option in `Sprite3D` for setting Blending mode of texture.

## Motivation:

I am creating open source fan-recreation of an old game, and that uses sprites like following:
![Light](https://github.com/godotengine/godot-proposals/assets/26041217/42951e87-b484-465e-ae8e-fa04af4f9e7d)

![image](https://github.com/godotengine/godot-proposals/assets/26041217/03936d69-55b0-4ada-adef-731dfecbaac5)

Which clearly is animated sprite yet is not a transparent texture while requiring it to be in 3D space. This will require addictive blending mode or one will need to manually edit texture, which may cause copyright issues.

So far I just created a mesh for each every UV positions and switching between meshes to mimic the expected behavior, therefore this suggestion has no real urgency.

![ezgif com-optimize (5)](https://github.com/godotengine/godot-proposals/assets/26041217/512d2eeb-457b-4f9f-b45c-2c39d4eced8a)

But would be nice to have blending option for any similar projects or for those who plans to use some other blending options while using `Sprite3D`'s powerful benefits.

**Comment by Calinou** [comment]
This can already be done by assigning a material override to Sprite3D (while keeping the texture in the texture slot to define its size), but it requires reassigning your albedo texture within the material.

It should be relatively straightforward to expose the blend mode as a Sprite3D property (also for Label3D). The same could also be done for the Disable Fog flag.

**Comment by jupiterbjy** [comment]
Ah mybad, I totally forgot `GeometryInstance3D` implements material overrides - this absolutely works, sorry for your time!

Since this is totally possible, I think it now is down to "Do we really need to expose blending mode?".

If every devs and users agrees that the term `Sprite`(regardless of dimensions) should carry the meaning of blending mode too, then would be better have it exposed.

But it seems to be that I am the only one (or maybe [two](https://www.reddit.com/r/godot/comments/xd96yt/how_to_make_sprite3d_emission_shader_just_for/)) that involves usecases requiring blending operations. So I'm not sure if I can continue insist on exposing it - since it's totally possible at current state without wacky workarounds.

Not to mention you devs are already seems to be under heavy workloads with preparing all the improvements and fixes - regardless of how easy exposing a superclass's property is.

So, will be closing this for now, instead I'll take my time studying engine & implementing this myself then open this as issue instead once ready. (After I manage to finish graduation project!)
